metis-enrichment/metis-enrichment-common/src/main/java/eu/europeana/enrichment/api/external/impl/EntityClientResolver.java

```diff
@@ -0,0 +1,174 @@
+                EnrichmentQuery clientRequest = requestParser.apply(input);
+                List<EnrichmentBase> enrichmentBaseList = executeEntityClientRequest(clientRequest);
+                if (!enrichmentBaseList.isEmpty()) {
+                    result.put(input, enrichmentBaseList);
+                }
+            }
+        }
+        return result;
+    }
+
+    private List<EnrichmentBase> executeEntityClientRequest(EnrichmentQuery clientRequest) {
+        // 1. get Entities
+        List<Entity> entities = getEntities(clientRequest);
+        if (!entities.isEmpty()) {
+            // 2. get the parent entities
+            List<Entity> parentEntities = new ArrayList<>();
+            entities.stream().forEach(entity -> {
+                if (entity.getIsPartOfArray() != null && EntityResolverUtils.isTextOrUriSearch(clientRequest)) {
+                    fetchParentEntities(entity, parentEntities);
+                }
+            });
+            // 3. add the parent entities
+            entities.addAll(parentEntities);
+            // 4. convert entity to EnrichmentBase
+            List<EnrichmentBase> enrichmentBaseList = convertToEnrichmentBase(entities);
+
+            EntityResolverUtils.failSafeCheck(entities.size(), enrichmentBaseList.size(), "Mismatch while converting the EM class to EnrichmentBase.");
+            return enrichmentBaseList;
+        }
+        return Collections.emptyList();
+    }
+
+
+    private List<Entity> getEntities(EnrichmentQuery entityClientRequest) {
+        try {
+            if (entityClientRequest.isReference()) {
+                return resolveReferences(entityClientRequest);
+            } else {
+                // if the enrich method returns more than 1 entity, then no enrichment should be (for now) considered.
+                // It works as if no entity was returned. The reason for this is that Metis does not have a disambiguation
+                // mechanism in place that can judge which entity should be chosen out of the list of options.
+                List<Entity> entities = entityClientApi.getEnrichment(entityClientRequest.getValueToEnrich(), entityClientRequest.getLanguage(), entityClientRequest.getType(), null) ;
+                if (entities.size() == 1) {
+                    return entities;
+                }
+            }
+        } catch (JsonProcessingException e) {
+            throw new UnknownException("Entity Client call to " + (entityClientRequest.isReference() ? "getEntityByUri" : "getSuggestions")
+                    + " failed for" + entityClientRequest + ".", e);
+        } catch (RuntimeException e) {
+            throw new UnknownException("Entity Client call failed for : " + entityClientRequest + ".", e);
+        }
+        return Collections.emptyList();
+    }
+
+    private List<Entity> resolveReferences(EnrichmentQuery entityClientRequest) {
+        if (entityClientRequest.getValueToEnrich().startsWith(EntityApiConstants.BASE_URL)) {
```

Comment:
So id checks are only considered values that start with **EntityApiConstants.BASE_URL**?
Metis does not have this distintion.
For example in metis when a client requests **resolveById** which does **not** start with **EntityApiConstants.BASE_URL** it will be checked as **resolveById** and not as **resolveByUri**.

With this change here it means that **resolveById** with values that do **not** start with **EntityApiConstants.BASE_URL** will be checked as **resolveByUri** which includes parent checking etc.
So a client is requesting a **resolveById** but gets essentially a **resolveByUri**

Is this really the intended behavior?

Also the **EntityApiConstants.BASE_URL**="http://data.europeana.eu". How about https? Perhaps we should be checking with a regex here to account for https too.

Reply:
So, you mean that the resolveByID can be requested without the full id like this '/agent/base/1' instead of this 'http://data.europeana.eu/agent/base/1' ??, 
I feel the simple solution should be to append the bases when we are requesting resolveByID. 
As it will be much faster to just fetch the ID directly rather than executing resolveByURI which is a two-step process

Reply:
After discussion:
- Semium requests are not expected anymore.
- The resolveById should return empty result if the id is not a europeana identifier and should not perform a resolveByUri.